File: sdkwork-backend-sdk-python/sdkwork_backend_sdk/models/plus_chat_message_vo.py
from dataclasses import dataclass
from typing import Optional, List, Dict, Any

@dataclass
class PlusChatMessageVO:
    """Chat Message Value Object - Represents a message in an AI conversation (user input or AI response)"""
    created_at: str = None
    updated_at: str = None
    id: int = None
    body: MessageBody = None
    role: str = None
    type: str = None
    status: str = None
    conversation_id: int = None
    conversation_uuid: str = None
    user_id: int = None
    sender_id: int = None
    sender: Participant = None
    receiver: Participant = None
    receiver_id: int = None
    group_id: int = None
    parent_message_id: int = None
    children: List[PlusChatMessageVO] = None
    metadata: Dict[str, Any] = None
    token_count: int = None
    send_at: str = None
    receive_at: str = None
    read_at: str = None
    processing_time: int = None
    is_error: bool = None
    error_code: str = None
    error_message: str = None
    model_id: int = None
    model: str = None
    temperature: float = None
    used_rag: bool = None
    chat_options: PlusChatOptions = None
    channel_id: str = None
    agent_id: int = None
    knowledge_base_id: int = None
    datasource_id: int = None
    agent_type: str = None
    agent_biz_type: str = None
    channel_msg_id: str = None
    channel_client_msg_id: str = None
    channel_msg_seq: int = None
    conversation_type: str = None
    feedback_metadata: FeedbackMetadata = None

File: sdkwork-backend-sdk-go/types/plus_chat_message_vo.go
package types

// Chat Message Value Object - Represents a message in an AI conversation (user input or AI response)
type PlusChatMessageVO struct {
	CreatedAt string `json:"createdAt"`
	UpdatedAt string `json:"updatedAt"`
	Id int `json:"id"`
	Body MessageBody `json:"body"`
	Role string `json:"role"`
	Type string `json:"type"`
	Status string `json:"status"`
	ConversationId int `json:"conversationId"`
	ConversationUuid string `json:"conversationUuid"`
	UserId int `json:"userId"`
	SenderId int `json:"senderId"`
	Sender Participant `json:"sender"`
	Receiver Participant `json:"receiver"`
	ReceiverId int `json:"receiverId"`
	GroupId int `json:"groupId"`
	ParentMessageId int `json:"parentMessageId"`
	Children []PlusChatMessageVO `json:"children"`
	Metadata map[string]interface{} `json:"metadata"`
	TokenCount int `json:"tokenCount"`
	SendAt string `json:"sendAt"`
	ReceiveAt string `json:"receiveAt"`
	ReadAt string `json:"readAt"`
	ProcessingTime int `json:"processingTime"`
	IsError bool `json:"isError"`
	ErrorCode string `json:"errorCode"`
	ErrorMessage string `json:"errorMessage"`
	ModelId int `json:"modelId"`
	Model string `json:"model"`
	Temperature float64 `json:"temperature"`
	UsedRag bool `json:"usedRag"`
	ChatOptions PlusChatOptions `json:"chatOptions"`
	ChannelId string `json:"channelId"`
	AgentId int `json:"agentId"`
	KnowledgeBaseId int `json:"knowledgeBaseId"`
	DatasourceId int `json:"datasourceId"`
	AgentType string `json:"agentType"`
	AgentBizType string `json:"agentBizType"`
	ChannelMsgId string `json:"channelMsgId"`
	ChannelClientMsgId string `json:"channelClientMsgId"`
	ChannelMsgSeq int `json:"channelMsgSeq"`
	ConversationType string `json:"conversationType"`
	FeedbackMetadata FeedbackMetadata `json:"feedbackMetadata"`
}
